backend/src/test/kotlin/cz/loono/backend/api/service/HealthcareProvidersServiceFilteringTest.kt

```diff
@@ -0,0 +1,124 @@
+        val healthcareProvidersService = HealthcareProvidersService(
+            healthcareProviderRepository,
+            healthcareCategoryRepository,
+            serverPropertiesRepository
+        )
+        val healthcareProvider = HealthcareProvider(
+            lat = 42.0,
+            lng = 50.6,
+            category = setOf(HealthcareCategory(value = CategoryValues.GENERAL_PRACTICAL_MEDICINE.value))
+        )
+
+        `when`(healthcareProviderRepository.findAll(any<PageRequest>()))
+            .thenReturn(
+                PageImpl(listOf(healthcareProvider))
+            )
+
+        val providers = healthcareProvidersService.findPage(0)
+
+        assert(providers.size == 1)
+        val filteredProvider = providers.first()
+        assert(filteredProvider.lat == healthcareProvider.lat)
+        assert(filteredProvider.lng == healthcareProvider.lng)
+        assert(filteredProvider.category.contains(CategoryValues.GENERAL_PRACTICAL_MEDICINE.value))
+    }
+
+    @Test
+    fun `missing coordinates`() {
+        val healthcareProvidersService = HealthcareProvidersService(
+            healthcareProviderRepository,
+            healthcareCategoryRepository,
+            serverPropertiesRepository
+        )
+        val healthcareProvider = HealthcareProvider(
+            category = setOf(HealthcareCategory(value = CategoryValues.GENERAL_PRACTICAL_MEDICINE.value))
+        )
+
+        `when`(healthcareProviderRepository.findAll(any<PageRequest>()))
+            .thenReturn(
+                PageImpl(listOf(healthcareProvider))
+            )
+
+        val providers = healthcareProvidersService.findPage(0)
+
+        assert(providers.isEmpty())
+    }
+
+    @Test
+    fun `missing category`() {
+        val healthcareProvidersService = HealthcareProvidersService(
+            healthcareProviderRepository,
+            healthcareCategoryRepository,
+            serverPropertiesRepository
+        )
+        val healthcareProvider = HealthcareProvider(
+            lat = 42.0,
+            lng = 50.6,
+            category = emptySet()
+        )
+
+        `when`(healthcareProviderRepository.findAll(any<PageRequest>()))
+            .thenReturn(
+                PageImpl(listOf(healthcareProvider))
+            )
+
+        val providers = healthcareProvidersService.findPage(0)
+
+        assert(providers.isEmpty())
+    }
+
+    @Test
+    fun `removing unwanted categories`() {
+        val healthcareProvidersService = HealthcareProvidersService(
+            healthcareProviderRepository,
+            healthcareCategoryRepository,
+            serverPropertiesRepository
+        )
+        val healthcareProvider = HealthcareProvider(
+            lat = 42.0,
+            lng = 50.6,
+            category = setOf(
+                HealthcareCategory(value = CategoryValues.GENERAL_PRACTICAL_MEDICINE.value),
+                HealthcareCategory(value = CategoryValues.PHARMACOLOGY.value)
```

Comment:
Veľmi spoliehaš na to, že nejaká neznáma hodnota je kdesi definovaná. Ideálne by to chcelo mať v teste definovaný list removedCategories, ale keďže je private, tak chápem :). To je len na zamyslenie, inak ok.